src/RandomForestBase.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+var DTClassfier = require('ml-cart').DecisionTreeClassifier;
+var DTRegression = require('ml-cart').DecisionTreeRegression;
+var Utils = require('./Utils');
+var Matrix = require('ml-matrix');
+
+var functions = {
+    'sqrt': Math.sqrt,
+    'log2': Math.log2
+};
+
+/**
+ * @class RandomForestBase
+ */
+class RandomForestBase {
+
+    /**
+     * Create a new base random forest for a classifier or regression model.
+     * @constructor
+     * @param {Object} options
+     * @param {Number|String} [options.maxFeatures] - the number of features used on each estimator.
+     *        * if is a String it support two methods to get the max features, "sqrt" or "log2" over all
+     *          sample features.
+     *        * if is an integer it selects maxFeatures elements over the sample features.
+     *        * if is a float between (0, 1), it takes the percentage of features.
+     * @param {Boolean} [options.replacement] - use replacement over the sample features.
+     * @param {Number} [options.seed] - seed for feature and samples selection, must be a 32-bit integer.
+     * @param {Number} [options.nEstimators] - number of estimator to use.
+     * @param {Object} [options.treeOptions] - options for the tree classifier, see [ml-cart]{@link https://mljs.github.io/decision-tree-cart/}
+     * @param {Boolean} [options.classifier] - boolean to check if is a classifier or regression model (used by subclasses).
+     */
+    constructor(options, model) {
+        if (options === true) {
+            this.options = model.options;
+            this.n = model.n;
+            this.indexes = model.indexes;
+
+            var Estimator = this.options.classifier ? DTClassfier : DTRegression;
+            this.estimators = model.estimators.map(est => Estimator.load(est));
+        } else {
+            if (options === undefined) options = {};
+            if (options.classifier === undefined) options.classifier = true;
+            if (options.nEstimators === undefined) options.nEstimators = 10;
+            if (options.replacement === undefined) options.replacement = true;
+            if (options.maxFeatures === undefined) options.maxFeatures = 'sqrt';
+
+            if (Utils.isString(options.maxFeatures) && functions[options.maxFeatures] === undefined) {
+                throw new RangeError('Not supported operation: ' + options.maxFeatures);
+            }
+
+            this.options = options;
+        }
+    }
+
+    /**
+     * Train the decision tree with the given training set and labels.
+     * @param {Matrix} trainingSet
+     * @param {Array} trainingValues
+     */
+    train(trainingSet, trainingValues) {
+        if (!Matrix.isMatrix(trainingSet)) {
```

Comment:
you can simply do `trainingSet = Matrix.checkMatrix(trainingSet)` (without condition, it checks if it is a matrix and converts it if not)